sema4ai/vscode-client/src/robo/agentPackage.ts

```diff
@@ -73,3 +77,80 @@ export const createAgentPackage = async (): Promise<void> => {
         window.showErrorMessage(errorMsg + " (see `OUTPUT > Sema4.ai` for more details).");
     }
 };
+
+export const packAgentPackage = async (): Promise<void> => {
+    const agentCli = getAgentCliLocation();
+    if (!agentCli) {
+        return;
+    }
+
+    let ws: WorkspaceFolder | undefined = await askForWs();
+    if (!ws) {
+        // Operation cancelled.
+        return;
+    }
+
+    const [rootPackageYaml, workspacePackages] = await Promise.all([
+        getPackageYamlNameFromDirectory(ws.uri),
+        getWorkspacePackages(),
+    ]);
+
+    if (!workspacePackages?.agentPackages?.length) {
+        window.showErrorMessage("No Agent Packages found in the workspace.");
+        return;
+    }
+
+    let packageInfo: LocalPackageMetadataInfo;
+    let actionPackageSelection: string | null = null;
+
+    /* If root level contains an agent-spec.yaml, there will be only one Agent Package. */
+    if (rootPackageYaml === PackageYamlName.Agent) {
+        packageInfo = workspacePackages.agentPackages[0];
+    } else {
+        actionPackageSelection = await window.showQuickPick(
+            workspacePackages.agentPackages.map((agentPackage) => agentPackage.name),
```

Comment:
Just noted that this could fail if the user has 2 agents with the same name... Given that the name is given from the user from the `agent-spec.yaml`, and copy-pasting can be common, I guess that maybe we should warn the user about the duplication as he won't know what he's selecting (and the `map` will make it so that we only show one randomly).

We should probably have a helper function for that and it should be used both here and the code from `actionPackage.ts` can be refactored to use that too.